TrackletAlgorithm/MemoryTemplate.h

```diff
@@ -130,24 +130,32 @@ class MemoryTemplate
 
   void print_mem(BunchXingT bx) const
   {
-	for (int i = 0; i <  nentries_[bx]; ++i) {
+	for (unsigned int i = 0; i <  nentries_[bx]; ++i) {
 	  std::cout << bx << " " << i << " ";
 	  print_entry(bx,i);
 	}
   }
 
   void print_mem() const
   {
-	for (int ibx = 0; ibx < (1<<NBIT_BX); ++ibx) {
-	  for (int i = 0; i < nentries_[ibx]; ++i) {
+	for (unsigned int ibx = 0; ibx < (1<<NBIT_BX); ++ibx) {
+	  for (unsigned int i = 0; i < nentries_[ibx]; ++i) {
 		std::cout << ibx << " " << i << " ";
 		print_entry(ibx,i);
 	  }
 	}
   }
 
   static constexpr int getWidth() {return DataType::getWidth();}
-  
+ 
+  std::string name_;
+  void setName(std::string name) { name_ = name;}
+  std::string const& getName() const { return name_;}
+
+  unsigned int iSector_;
+  void setSector(unsigned int iS) { iSector_ = iS;}
+  unsigned int getSector() const { return iSector_;}
```

Comment:
@carriganm95 These lines and the similar ones in the other template classes cause the C/RTL cosimulation to fail for most of the processing modules. After playing around a bit, I think the problem is having data members (`name_` and `iSector_` in this case) that do not exist during synthesis, but do exist when the C++ test bench is run. Somehow this difference in the class members screws up the cosimulation.

I would suggest protecting these lines with `#ifdef CMSSW_GIT_HASH` instead of `#ifndef __SYNTHESIS__`. That should allow the cosimulation to run successfully again.

Reply:
Thanks. I noticed the fail yesterday but hadn’t figured out what went wrong. I will make this change.